tower-reconnect/src/lib.rs

```diff
@@ -22,15 +22,14 @@ pub enum Error<T, U> {
     NotReady,
 }
 
-pub struct ResponseFuture<T>
-where T: NewService
+pub struct ResponseFuture<T, R>
+where T: NewService<R>
```

Comment:
Again, the `T: NewService` bound had to be maintained to use the `T::Service` associated type, so an additional type parameter was added.